File: 1-easy/problems/2855-minimum-right-shifts-to-sort-the-array/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def minimumRightShifts(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i = next((i for i in range(len(nums)) if not nums[i] < nums[(i+1)%len(nums)]), len(nums))
        j = next((j for j in range(i+1, len(nums)) if not nums[j%len(nums)] < nums[(j+1)%len(nums)]), len(nums))
        return len(nums)-(i+1) if j == len(nums) else -1

File: 1-easy/problems/2855-minimum-right-shifts-to-sort-the-array/solution.ts
function minimumRightShifts(nums: any): boolean | number | string | any {
  """
  :type nums: List[int]
  :rtype: int
  """
  i = next((i for i in Array.from({length: (nums}, (_, i) => i).length) if not nums[i] < nums[(i+1)%(nums).length]), (nums).length)
  j = next((j for j in Array.from({length: i+1, (nums}, (_, i) => i).length) if not nums[j%(nums).length] < nums[(j+1)%(nums).length]), (nums).length)
  return (nums).length-(i+1) if j == (nums).length else -1;
}

export { minimumRightShifts };
